src/pygambit/game.pxi

```diff
@@ -336,6 +337,35 @@ class Game:
         g.title = title
         return g
 
+    def to_arrays(self, dtype: typing.Type = Rational) -> typing.List[np.array]:
+        """Generate the payoff tables for players represented as numpy arrays.
+        Parameters
+        ----------
+        dtype : type
+            The type to which payoff values will be converted and
+            the resulting arrays will be of that dtype
+
+        Returns
+        -------
+        list of np.array
+
+        See Also
+        --------
+        from_arrays : Create game from list-like of array-like
+        """
+        arrays = []
+
+        shape = tuple(len(player.strategies) for player in self.players)
+        for player in self.players:
+            array = np.zeros(shape=shape, dtype=object)
+            for profile in itertools.product(*(range(s) for s in shape)):
+                try:
+                    array[profile] = dtype(self[profile][player])
+                except ValueError:
+                    print("Unsupported type of payoff")
```

Comment:
The behaviour of this is that it actually suppresses the error.  Yes it prints something, but then the code continues to run and returns arrays, but those arrays will be wrong.

The function needs to raise an exception, but our goal here is to make it raise an exception that will be more useful to users to understand the problem.  I would suggest:

```
raise ValueError(f"Payoff '{self[profile][player]}' cannot be converted to requested type '{dtype}'") from None
```

With this, instead the sequence in the above comment behaves like this:

```
In [3]: g.to_arrays(dict)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[3], line 1
----> 1 g.to_arrays(dict)

File ~/git/gambit/src/pygambit/game.pxi:365, in pygambit.gambit.Game.to_arrays()
    363             array[profile] = dtype(self[profile][player])
    364         except (ValueError, TypeError):
--> 365             raise ValueError(f"Payoff '{self[profile][player]}' cannot be converted to requested type '{dtype}'") from None
    366     arrays.append(array)
    367 return arrays

ValueError: Payoff '1' cannot be converted to requested type '<class 'dict'>'
```

Which means that we're always raising a `ValueError` (irrespective of what the dtype raises), and then giving a useful error message to the user to understand the problem.

The effect of the `from None` is that it suppresses the original exception; if we don't include it the user gets two exception stack traces (one from the original exception and one from ours).  In most cases, the original stack trace isn't going to be interesting, because the point is that the requested type can't represent the payoff.